Test for MathJax rendering in training e2e tests

dmnisson opened the pull request:
Links
-----
- Issue: #333 
- Server repo PR: UPchieve/server#249 

Related PR: #337 

Description
-----------
- The `QuizView` is updated to typeset MathJax equations on question text and all possible answers on quiz pages themselves and on the review page.
- A new test case is added to `tests/e2e/specs/training-tests.js` to check that any MathJax equations are being rendered in quiz question text, all possible answers, and on the quiz review page.

Developer self-review checklist
-------------------------------
- [x] Potentially confusing code has been explained with comments
- [x] No warnings or errors have been introduced; all known error cases have been handled
- [x] Any appropriate documentation (within the code, README.md, docs, etc) has been updated
- [x] All edge cases have been addressed

jkeat commented:
@dmnisson could you take a look at the merge conflicts in `specs/training-tests.js`?

jkeat commented:
@dmnisson I'm having trouble getting these new tests to pass. Thought it might have to do with the review page having duplicate IDs for the answer elements (every question has an `#answer-a`, `#answer-b`, etc), but after fixing that, they're still failing.

I'll push up the answer ID change because it's still probably a good idea to avoid duplicate element IDs.

dmnisson commented:
> @dmnisson I'm having trouble getting the new test to pass. Thought it might have to do with the review page having duplicate IDs for the answer elements (every question has an `#answer-a`, `#answer-b`, etc), but after fixing that, they're still failing.
> 
> I'll push up the answer ID change because it's still probably a good idea to avoid duplicate element IDs.

I've run these tests a couple of times, found what I think were the problems, and corrected them. In the Cypress headless mode, the browser version (Electron 61) does not support `Array.prototype.flatMap`, so the MathJax actually was not being rendered on the review page in Electron. 

Also, the `rerenderMathJaxElements` code was queuing `Typeset` actions for all of the question text and answer elements on the review page, including those that did not actually have MathJax in them, while the tests started on the first element that had MathJax. I changed the `rerenderMathJaxElements` function to queue `Typeset` actions only for those elements that have backslashes in them, and also set the timeouts to wait longer for the MathJax rendering. The tests now seem to pass.
